libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -156,22 +156,42 @@ void ContractLevelChecker::findDuplicateDefinitions(map<string, vector<T>> const
 	}
 }
 
-void ContractLevelChecker::checkAbstractFunctions(ContractDefinition const& _contract)
+void ContractLevelChecker::checkAbstractDefinitions(ContractDefinition const& _contract)
 {
+	auto hasSameSignature = [](OverrideProxy const& a, OverrideProxy const& b)
+	{
+		auto _a = a.overrideComparator();
+		auto _b = b.overrideComparator();
+
+		if (_a.name != _b.name)
+			return false;
+
+		// Dealing with modifiers. Modifiers with same name and different
+		// signature are already checked.
+		if (!_a.functionKind && !_b.functionKind)
+			return true;
+
+		if (_a.functionKind != _b.functionKind)
+			return false;
+
+		return (_a.parameterTypes == _b.parameterTypes);
+	};
+
 	// Mapping from name to function definition (exactly one per argument type equality class) and
 	// flag to indicate whether it is fully implemented.
-	using FunTypeAndFlag = std::pair<FunctionTypePointer, bool>;
-	map<string, vector<FunTypeAndFlag>> functions;
+	using ProxyAndFlag = std::pair<OverrideProxy, bool>;
+	map<string, vector<ProxyAndFlag>> proxies;
```

Comment:
Can you try to use
```
map<string, set<OverrideProxy, OverrideProxy::CompareBySignature>> proxies;
```
here?

And then `registerProxy` becomes:
```
auto registerProxy = [&](OverrideProxy const& _overrideProxy)
{
  auto& overloads = proxies[_overrideProxy.name()];
  if (overloads.count(_overrideProxy)) {
    // replace the existing proxy if that does not implement but the new one implements
  }
  else
    overloads.insert(_overrideProxy);
}
```

Reply:
I just implemented it. Instead of `map<string, set<OverrideProxy, OverrideProxy::CompareBySignature>>`, I used just `set<OverrideProxy, OverrideProxy::CompareBySignature>`.